batchread/batchread.go

```diff
@@ -112,8 +115,11 @@ func readEntriesBatch(inputBatchChan chan readMicroBatch) {
 		for rows.Next() {
 			var pk, payload string
 			err = rows.Scan(&pk, &payload)
+			panicIfError(err)
 			rowsMap[pk] = payload
 		}
+		panicIfError(err)
+		rows.Close()
```

Comment:
For loop closes rows automatically (unless you break - which I dont)

Reply:
True, not use to panic I guess - because we should have handled the case of error which will not enter the loop.

Reply:
btw, it means there is no need to switch case where we set 'err' in the reply - we only set not found - all other errors we panic.

Reply:
Right, but I think it should stay to show it is possible to pass other errors to he reader

Reply:
so, keep the call to close so it will not be an optional resource leak in case they do handle the errors.